csharp_dotnetcore/14.NLP-With-Dispatch/README.md

```diff
@@ -0,0 +1,146 @@
+
+Note: You can create the LUIS applications in one of the [LUIS authoring regions](https://docs.microsoft.com/en-us/azure/cognitive-services/LUIS/luis-reference-regions). 
+You can use a different region (westus or westeurope or australiaeast) via https://**LUIS-Authoring-Region**.api.cognitive.microsoft.com/luis/api/v2.0 in the commands above.
+
+### Train and publish the LUIS models 
+You need to train and publish the LUIS models that were created for this sample to work. You can do so using the following CLI commands
+
+```bash
+> msbot get service --name homeautomation.luis | luis train version --wait --stdin
+> msbot get service --name homeautomation.luis | luis publish version --stdin
+> msbot get service --name weather.luis | luis train version --wait --stdin
+> msbot get service --name weather.luis | luis publish version --stdin
+```
+
+### Configure QnA Maker service
+To create a new QnA Maker application for the bot, 
+- Follow instructions [here](https://docs.microsoft.com/en-us/azure/cognitive-services/qnamaker/how-to/set-up-qnamaker-service-azure) to create a new QnA Maker Azure resource.
+- Navigate to your QnA Maker resource -> keys and copy the subscription key
+
+To create the QnA Maker application and update the .bot file with the QnA Maker configuration,  
+- Open a terminal
+- Navigate to samples\14.nlp-with-dispatch
+- Run the following command
+```bash
+> ludown parse toqna --in dialogs/qna/resources/sample-qna.lu -o cognitiveModels -n sample.qna
+
+> qnamaker create kb --in cognitiveModels\sample.qna --subscriptionKey <YOUR-QNA-SUBSCRIPTION-KEY> --msbot | msbot connect qna --stdin --name sample.qna
+```
+### Train and publish the QnA Maker KB
+You need to train and publish the QnA Maker Knowledge Bases that were created for this sample to work. You can do so using the following CLI commands
+
+```bash
+> msbot get service --name sample.qna | qnamaker publish kb --stdin
+```
+
+### Configure the Dispatch application
+[Dispatch](https://github.com/Microsoft/botbuilder-tools/tree/master/packages/Dispatch) is a CLI tool that enables you to create a dispatch NLP model across the different LUIS applications and/ or QnA Maker Knowledge Bases you have for your bot. For this sample, you would have already created 2 LUIS applications (Home Automation and Weather) and one QnA Maker Knowledge base. 
+
+To create a new dispatch model for these services and update the .bot file configuration, in a terminal,
+- Navigate to samples\14.nlp-with-dispatch
+- Run the following command
+```bash
+> dispatch create -b nlp-with-dispatch.bot | msbot connect dispatch --stdin --name bot-dispatch
+```
+### Securing keys in your .bot file
+Since your .bot file contains service IDs, subscription and authoring keys, its best to encrypt them. To encrypt the .bot file, you can run
```

Comment:
>IDs [](start = 38, length = 3)

casing. Ids. I think. I'm on the fence about this one.

Reply:
Both sound weird.

Reply:
* Both look weird when read, yet they sound so natural when spoken :s